edx_exams/apps/api/v1/views.py

```diff
@@ -271,3 +276,106 @@ class ProctoringProvidersView(ListAPIView):
     model = ProctoringProvider
     serializer_class = ProctoringProviderSerializer
     queryset = ProctoringProvider.objects.all()
+
+
+class ExamAccessTicketsView(APIView):
+    """
+    View to create signed exam access tickets for a specific user and exam.
+
+    Given an exam_id and user (in request), this view will either grant access
+    as an exam access ticket or not grant access. Access is granted if there is an
+    existing exam attempt or if the exam is past its due date.
+
+    HTTP GET
+    Provides an Exam Access Ticket as a cookie if access is granted.
+    **GET data Parameters**
+        * exam_id: This is the id of the exam that the user is requesting access to
+    **Exceptions**
+        * HTTP_403_FORBIDDEN
+    """
+
+    authentication_classes = (JwtAuthentication,)
+    permission_classes = (StaffUserOrReadOnlyPermissions,)
+
+    @classmethod
+    def get_expiration_window(cls, exam_attempt, default_exp_seconds):
+        """
+        Set exam access ticket expiration window.
+
+        Use default exp window or the time to the end of the exam attempt,
+        whichever is shorter.
+        """
+        secs_remaining = get_exam_attempt_time_remaining(exam_attempt)
+
+        # If not past due and attempt time remaining is less than default, set window to remaining
+        if 0 <= secs_remaining < default_exp_seconds:
+            exp_window = secs_remaining
+        else:
+            exp_window = default_exp_seconds
+
+        return exp_window
+
+    @classmethod
+    def get_response(cls, exam, user):
+        """
+        Get response, with exam access ticket if access granted.
+
+        403 error if access is not granted.
+        """
+        claims = {"course_id": exam.course_id, "content_id": exam.content_id}
+        expiration_window = 60
+        exam_attempt = ExamAttempt.get_current_exam_attempt(user, exam)
+
+        data = {"detail": "Exam access ticket not granted"}
+        grant_access = False
+        response_status = status.HTTP_403_FORBIDDEN
+
+        # If exam attempt exists for user, then grant exam access
+        if exam_attempt is not None:
+            # If no due date, grant access if attempt started
+            if exam.due_date is None:
+                if exam_attempt.status == ExamAttemptStatus.started:
```

Comment:
if there's no due date don't we still need to get_expiration_window since that would take into account the time remaining on the attempt which might be less than the default expiration_window

Reply:
Ah good point. Just fixed.